main.cpp

```diff
@@ -161,22 +163,35 @@ int main (int argc, char* argv[])
     }
 
     WallpaperEngine::Core::CProject* project = WallpaperEngine::Core::CProject::fromFile (project_path);
-    WallpaperEngine::Render::CScene* scene = new WallpaperEngine::Render::CScene (project, IrrlichtContext);
+    WallpaperEngine::Render::CWallpaper* wallpaper;
+
+    if (strcmp (project->getType ().c_str (), "scene") == 0)
```

Comment:
Do not use C functions for strings, *type can be directly compared with "==" to "scene", it'll take care of the casting to std::string and call the proper function for comparing them.